components/profile-controller/pkg/controller/profile/profile_controller.go

```diff
@@ -166,13 +174,14 @@ func (r *ReconcileProfile) Reconcile(request reconcile.Request) (reconcile.Resul
 
 	roleBinding := &rbacv1.RoleBinding{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "default",
+			Name:      "namespaceAdmin",
 			Namespace: instance.Name,
 		},
+		// Use default ClusterRole 'admin' for profile/namespace owner
 		RoleRef: rbacv1.RoleRef{
 			APIGroup: "rbac.authorization.k8s.io",
-			Kind:     "Role",
```

Comment:
Why do we bind clusterrole admin to the user?

Reply:
Creating a roleBinding in target namespace will grant user namespace-scoped permission.

Reply:
could you add a comment explaining that clusterrole can be to namespaced roleref and only has namespaced access?